yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,295 @@
+        # values.
+        #
+        self.length_unit = self.quan(1.0, "cm")
+        self.mass_unit = self.quan(1.0, "g")
+        self.time_unit = self.quan(1.0, "s")
+        self.time_unit = self.quan(1.0, "s")
+        self.velocity_unit = self.quan(1.0, "cm/s")
+        self.magnetic_unit = self.quan(1.0, "gauss")
+
+    def _parse_parameter_file(self):
+        with h5py.File(self.parameter_filename, "r") as f:
+            # Reads in simulation time, number of dimensions and shape
+            self.current_time = f["mesh"]["time"][()]
+            self.dimensionality = 3
+            self.domain_dimensions = f["mesh"]["array_dimensions"][()]
+
+            index_filenames = _find_files(self.parameter_filename)
+            if any(
+                "grid_2" in file for file in index_filenames
+            ):  # Checks for Yin-Yang and changes to cartesian
+                self.geometry = "cartesian"
+                self._periodicity = (False, False, False)
+                dre = [
+                    f["mesh"]["x_ef"][(-1)],
+                    f["mesh"]["x_ef"][(-1)],
+                    f["mesh"]["x_ef"][(-1)],
+                ]
+                dle = -1 * np.array(dre)
+                self.domain_dimensions = np.array(
+                    [
+                        2 * f["mesh"]["array_dimensions"][0],
+                        2 * f["mesh"]["array_dimensions"][0],
+                        2 * f["mesh"]["array_dimensions"][0],
+                    ]
+                )
+            else:
+                self.geometry = "spherical"  # Uses default spherical geometry
+                self._periodicity = (False, False, True)
+                dle = [
+                    f["mesh"]["x_ef"][(0)],
+                    f["mesh"]["y_ef"][(0)],
+                    f["mesh"]["z_ef"][(0)],
+                ]
+                if self.domain_dimensions[2] > 2:
+                    dre = [
+                        f["mesh"]["x_ef"][(-3)],
+                        f["mesh"]["y_ef"][(-1)],
+                        f["mesh"]["z_ef"][(13)],
+                    ]
+                else:
+                    dre = [
+                        f["mesh"]["x_ef"][(-1)],
+                        f["mesh"]["y_ef"][(-1)],
+                        f["mesh"]["z_ef"][(-1)],
+                    ]
+            # Sets left and right bounds based on earlier definitions
+            self.domain_right_edge = np.array(dre)
+            self.domain_left_edge = np.array(dle)
+
+        self.cosmological_simulation = 0.0  # Chimera is not a cosmological simulation
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+        try:
+            fileh = HDF5FileHandler(args[0])
+            if "fluid" in fileh and "agr_c" in fileh["fluid"].keys():
```

Comment:
thought more about this -- the intersection of "fluid" and the "arg_c" key is probably sufficient. But if the output happens to have metadata that is clearly Chimera-specific, I do think it'd be worth adding a narrower check. If not, this is OK!

Reply:
While newer iterations of the code do output metadata, it's not very Chimera specific nor reliably invariant from run to run, not to mention older datasets do not have it. I looked into finding better checks, but this one seemed like the best intersection of reliable presence in Chimera files and somewhat unlikelihood to be found elsewhere. What would be possible is to change the check to look for multiple Chimera filed names, rather than just "agr_c", to decrease the chance of accidentally misidentifying a non-chimera file.

Reply:
Ya, adding another field to check for is a good idea I think! But only if you're already changing other things -- it'd be an easy thing to fix in the unlikely event that there are conflicts.

Reply:
I've added a second field check, let me know if the field combination looks unique enough.